src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/WorkspaceProjectStateChangeDetector.cs

```diff
@@ -292,23 +302,27 @@ private async Task Workspace_WorkspaceChangedAsync(WorkspaceChangeEventArgs e, C
                 Debug.Fail("WorkspaceProjectStateChangeDetector.Workspace_WorkspaceChanged threw exception:" +
                     Environment.NewLine + ex.Message + Environment.NewLine + "Stack trace:" + Environment.NewLine + ex.StackTrace);
             }
+        }
 
-            static bool IsRazorFileOrRazorVirtual(Document document)
-            {
-                // Using EndsWith because Path.GetExtension will ignore everything before .cs
-                // Using Ordinal because the SDK generates these filenames.
-                // Stll have .cshtml.g.cs and .razor.g.cs for Razor.VSCode scenarios.
-                return document.FilePath.EndsWith(".cshtml.g.cs", FilePathComparison.Instance) ||
-                    document.FilePath.EndsWith(".razor.g.cs", FilePathComparison.Instance) ||
-                    document.FilePath.EndsWith(".razor", FilePathComparison.Instance) ||
+        private bool IsRazorFileOrRazorVirtual(Document document)
+        {
+            if (document.FilePath is null)
+                return false;
 
-                    // VSCode's background C# document
-                    document.FilePath.EndsWith("__bg__virtual.cs", StringComparison.Ordinal);
-            }
+            // Using EndsWith because Path.GetExtension will ignore everything before .cs
+            return document.FilePath.EndsWith(_languageServerFeatureOptions.CSharpVirtualDocumentSuffix, FilePathComparison.Instance) ||
```

Comment:
This is the fix.

This means that files ending in ".ide.g.cs" will now trigger a recompile. We could arguably make this check ".razor.ide.g.cs" and ".cshtml.ide.g.cs", but I don't personally think its worth worrying about.

Reply:
All of the other changes in this PR are just to avoid hardcoding ".ide.g.cs" so that hopefully we avoid this regression in future. Probably also could have made this whole method just `.Contains("razor")` and put up with the false positives 😛